src/components/refactor/dialogs/generator-scaling/variation/variation-form.js

```diff
@@ -0,0 +1,170 @@
+                        const filterWithMetadata = results.find(
+                            (f) => f.elementUuid === filter.id
+                        );
+                        if (filterWithMetadata) {
+                            return {
+                                [ID]: filterWithMetadata.elementUuid,
+                                [NAME]: filterWithMetadata.elementName,
+                                [SPECIFIC_METADATA]:
+                                    filterWithMetadata.specificMetadata,
+                            };
+                        }
+                        return filter;
+                    });
+                    setValue(filterFieldName, newFilters);
+                })
+                .catch((errorMessage) => {
+                    snackError({
+                        messageTxt: errorMessage,
+                        headerId: 'GeneratorScalingError',
+                    });
+                });
+        },
+        [filterFieldName, filters, setValue, snackError]
+    );
+
+    useEffect(() => {
+        // When editing the modification, filters does not have specific metadata which contain filter type
+        // If variation mode is STACKING_UP or VENTILATION, all filters types must be 'explicit naming'
+        if (
+            (variationMode === VARIATION_MODES.STACKING_UP.id ||
+                variationMode === VARIATION_MODES.VENTILATION.id) &&
+            filters.length > 0
+        ) {
+            // collect all filters without metadata
+            const filtersWithoutMetadata = filters.filter((filter) => {
+                return !filter?.specificMetadata;
+            });
+
+            if (filtersWithoutMetadata.length > 0) {
+                updateMetadata(filtersWithoutMetadata);
+            }
+        }
+    }, [variationMode, filters, updateMetadata]);
+
+    const itemFilter = useCallback(
+        (value) => {
+            if (value?.type === elementType.FILTER) {
+                if (variationMode === VARIATION_MODES.STACKING_UP.id) {
+                    return value?.specificMetadata?.type === IDENTIFIER_LIST;
+                }
+
+                if (variationMode === VARIATION_MODES.VENTILATION.id) {
+                    return (
+                        value?.specificMetadata?.type === IDENTIFIER_LIST &&
+                        value?.specificMetadata?.filterEquipmentsAttributes?.every(
+                            (fil) => !!fil.distributionKey
+                        )
+                    );
+                }
+            }
+
+            return true;
+        },
+        [variationMode]
+    );
+
+    const filtersField = (
+        <DirectoryItemsInput
+            name={`${name}.${index}.${FILTERS}`}
+            equipmentTypes={GENERATORS}
+            elementType={elementType.FILTER}
+            label={'filter'}
+            titleId={'FiltersListsSelection'}
+            itemFilter={itemFilter}
+            optional={false}
```

Comment:
optional status should be extracted from yup schema to avoid code duplication / desync, take a look at the other inputs